src/charts/loading/LoadingContainer.js

```diff
@@ -0,0 +1,73 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+export const loadingContainerWrapper = ({data, chart, shouldShowLoadingState}, container) => {
+    if (shouldShowLoadingState) {
+        container = (
+            <LoadingContainer
+                data={data}
+                loadingState={chart.loading()}
+            >
+                {container}
+            </LoadingContainer>
+        );
+    }
+    return container;
+}
+
+const toggleLoading = (state) => ({loading: !state.loading});
+
+export default class LoadingContainer extends PureComponent {
+    static propTypes = {
+        children: PropTypes.element,
+        data: PropTypes.oneOfType(
+            [
+                PropTypes.object,
+                PropTypes.array,
+            ]
+        ),
+        loadingState: PropTypes.string,
+    }
+
+    state = {
+        loading: true,
+    }
+
+    componentWillMount() {
+        if (this.props.data !== null) {
+            this.setState(toggleLoading);
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.data === null && nextProps.data !== null) {
+            this.setState(toggleLoading);
+        }
+    }
+
+    render() {
+        const {
+            children,
+            loadingState,
+        } = this.props;
+
+        let className = 'loading-container';
+
+        if (this.state.loading) {
+            className += ' loading-container--loading';
+        }
```

Comment:
nit: we should consider introducing the classnames module here to avoid having to do this. not a big deal though since there's so few classes

Reply:
Good idea, that would make this cleaner.

Reply:
Nvm, we need to use inline styles anyway